File: ABC/ABC235/C/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;
class Program
{
    static void Main(string[] args)
    {
        var m = new Dictionary<int, List<int>>();
        var line1 = Console.ReadLine().Split(" ").Select(int.Parse).ToArray();
        var n = line1[0];
        var q = line1[1];
        var a = Console.ReadLine().Split(" ").Select(int.Parse).ToArray();
        for (var i = 0; i < n; i++) {
            if (!m.ContainsKey(a[i]))
            {
                m.Add(a[i], new List<int>());
                m[a[i]].Add(i + 1);
            }
            else
            {
                m[a[i]].Add(i + 1);
            }
        }
        for (var j = 0; j < q; j++)
        {
            var line = Console.ReadLine().Split(" ").Select(int.Parse).ToArray();
            var x = line[0];
            var k = line[1];
            if (!m.ContainsKey(x) || m[x].Count < k) {
                Console.WriteLine(-1);
            } else {
                Console.WriteLine(m[x][k - 1]);
            }
        }
    }
}

File: ABC/ABC235/C/dart/program.dart
import 'dart:convert';
import 'dart:io';
import 'dart:async';

Future<void> main(List<String> args) async {
  var line1 = stdin.readLineSync()?.split(" ") ?? [];

  var m = Map<int, List<int>>();
  var n = int.parse(line1[0]);
  var q = int.parse(line1[1]);

  var isFirst = true;
  List<int> a;

  var lines = stdin.transform(utf8.decoder).transform(const LineSplitter());
  await for (var line in lines) {
    if (isFirst) {
      a = line.split(" ").map(int.parse).toList();
      isFirst = false;
      for (var i = 0; i < n; i++) {
        if (!m.containsKey(a[i])) {
          m[a[i]] = [];
        }
        m[a[i]]?.add(i + 1);
      }
    } else {
      var s = line.split(" ");
      var x = int.parse(s[0]);
      var k = int.parse(s[1]);
      var l = m[x];
      if (l == null || l.length < k) {
        print(-1);
      } else {
        print(l[k - 1]);
      }
    }
  }
}
